tensorflow/lite/delegates/webnn/webnn_delegate.cc

```diff
@@ -2534,6 +2592,52 @@ class Subgraph {
     return kTfLiteOk;
   }
 
+  static TfLiteStatus VisitTransposeNode(
+      const emscripten::val& builder, TfLiteContext* logging_context,
+      int node_index, TfLiteNode* node, const TfLiteTensor* tensors,
+      std::unordered_map<int, emscripten::val>& webnn_operands,
+      const bool detect_supported_op) {
+    TF_LITE_ENSURE_STATUS(
+        CheckNumInputsAndOutputs(logging_context, node, 2, 1, node_index));
+
+    const int input_tensor_id = node->inputs->data[0];
+    const TfLiteTensor& input_tensor = tensors[input_tensor_id];
+    TF_LITE_ENSURE_STATUS(CheckTensorNonDynamicAllocation(
+        logging_context, input_tensor, input_tensor_id, node_index));
+
+    const TfLiteTensor& perm_tensor = tensors[node->inputs->data[1]];
+    TF_LITE_ENSURE_STATUS(CheckTensorStaticAllocation(
+        logging_context, perm_tensor, node->inputs->data[1], node_index));
+
+    const int32_t* perm_data =
+        reinterpret_cast<const int32_t*>(perm_tensor.data.data);
+
+    const int output_tensor_id = node->outputs->data[0];
+    const TfLiteTensor& output_tensor = tensors[output_tensor_id];
+    TF_LITE_ENSURE_STATUS(CheckTensorNonDynamicAllocation(
+        logging_context, output_tensor, output_tensor_id, node_index));
+    const int dims_count = NumDimensions(&input_tensor);
+    std::vector<int32_t> perm;
```

Comment:
WebNN requires the unsigned integer permutation. Does TFLite support negative value? Should the signed integer be converted to unsigned?

Reply:
Thanks @huningxin, I've updated the code as we discussed, PTAL again. Thanks!